src/browser-polyfill.js

```diff
@@ -382,36 +396,96 @@ if (typeof browser === "undefined") {
           return false;
         }
 
+        // A small helper to send the message if the promise resolves
+        // and an error if the promise rejects (a wrapped sendMessage has
+        // to translate the message into a resolved promise or a rejected
+        // promise).
+        const sendPromisedResult = (promise) => {
+          promise.then(msg => {
+            // send the message value.
+            sendResponse(msg);
+          }, error => {
+            console.error(error);
+            // Send a JSON representation of the error if the rejected value
+            // is an instance of error, or the object itself otherwise.
+            const isErrorInstance = error instanceof Error;
+            sendResponse({
+              __webExtensionPolyfillReject__: {isErrorInstance},
+              error: !isErrorInstance ? error : {
+                message: error.message,
+                fileName: error.fileName,
+                lineNumber: error.lineNumber,
+              },
+            });
```

Comment:
I personally prefer `__mozWebExtPolyfill_isSerializedErrorObject` over `__webExtensionPolyfillReject__`, [the reasons are explained here](https://github.com/mozilla/webextension-polyfill/pull/97#discussion_r178991820):
> Perhaps something like `__mozWebExtPolyfill_isSerializedErrorObject` could work, because `isInstanceOfError` is likely to be used by something.
>
> The double-`_` prefix is used since it’s somewhat common practice to use `__comment` in place of real comments in JSON, and a single-`_` prefix already denotes an internal varible.
>
> The [`moz` prefix](https://developer.mozilla.org/en-US/docs/Glossary/Vendor_Prefix#Property_and_method_prefixes) is also used by non-standard Mozilla properties on Web APIs, and `WebExtPolyfill` identifies this polyfill.